src/main/java/com/epam/aidial/core/service/PublicationService.java

```diff
@@ -473,6 +477,24 @@ private String encodeReviewBucket(ResourceDescription bucket, String id) {
         return encryption.encrypt(path);
     }
 
+    private Map<String, List<Rule>> getCachedRules() {
+        ResourceItemMetadata meta = resources.getResourceMetadata(PUBLIC_RULES);
+        long key = (meta == null) ? Long.MIN_VALUE : meta.getUpdatedAt();
+        Pair<Long, Map<String, List<Rule>>> current = cachedRules.get();
+
+        if (current == null || current.getKey() != key) {
+            Pair<ResourceItemMetadata, String> resource = resources.getResourceWithMetadata(PUBLIC_RULES);
+            Pair<Long, Map<String, List<Rule>>> next = (resource == null)
+                    ? Pair.of(Long.MIN_VALUE, decodeRules(null))
+                    : Pair.of(resource.getKey().getUpdatedAt(), decodeRules(resource.getValue()));
+
+            cachedRules.compareAndSet(current, next);
```

Comment:
you should evaluate a result of the CAS operation to make sure it's successful

Reply:
Not needed.